misc/cron/archive.php

```diff
@@ -59,41 +52,9 @@
     array_unshift($_SERVER['argv'], $script);
 
     $console->run();
-} else { // if running via web request, use CronArchive directly
-
-    if (Piwik\Common::isPhpCliMode()) {
-        // We can run the archive in CLI with `php-cgi` so we have to configure the container/logger
-        // just like for CLI
-        $environment = new \Piwik\Application\Environment('cli');
-        $environment->init();
-
-        /** @var ConsoleHandler $consoleLogHandler */
-        $consoleLogHandler = StaticContainer::get('Symfony\Bridge\Monolog\Handler\ConsoleHandler');
-        $consoleLogHandler->setOutput(new ConsoleOutput(OutputInterface::VERBOSITY_VERBOSE));
-    } else {
-        // HTTP request: logs needs to be dumped in the HTTP response (on top of existing log destinations)
-        $environment = new \Piwik\Application\Environment(null);
-        $environment->init();
-
-        /** @var \Monolog\Logger $logger */
-        $logger = StaticContainer::get('Psr\Log\LoggerInterface');
-        $handler = new StreamHandler('php://output', Logger::INFO);
-        $handler->setFormatter(StaticContainer::get('Piwik\Plugins\Monolog\Formatter\LineMessageFormatter'));
-        $logger->pushHandler($handler);
-    }
-
-    $archiver = new Piwik\CronArchive();
-
-    if (!Piwik\Common::isPhpCliMode()) {
-        $token_auth = Piwik\Common::getRequestVar('token_auth', '', 'string');
-
-        if (!$archiver->isTokenAuthSuperUserToken($token_auth)) {
-            die('<b>You must specify the Super User token_auth as a parameter to this script, eg. <code>?token_auth=XYZ</code> if you wish to run this script through the browser. </b><br>
-                    However it is recommended to run it <a href="http://piwik.org/docs/setup-auto-archiving/">via cron in the command line</a>, since it can take a long time to run.<br/>
-                    In a shell, execute for example the following to trigger archiving on the local Piwik server:<br/>
-                    <code>$ /path/to/php /path/to/piwik/console core:archive --url=http://your-website.org/path/to/piwik/</code>');
-        }
-    }
+} else { // if running via web request, use CoreAdminHome.runCronArchiving method
+    $_GET['module'] = 'API';
+    $_GET['method'] = 'CoreAdminHome.runCronArchiving';
 
-    $archiver->main();
+    require_once PIWIK_INCLUDE_PATH . "/index.php";
```

Comment:
> why not keep calling the archiver directly? I feels cleaner to me than to simulate a HTTP request by mocking global variables

Mocking global variables? Am I doing that? The script calls CronArchive directly after doing a hacky authentication check. If it requires authentication, we should use the API method no? And having an API method for web accessible functionality is better than a custom script no?

>  it shouldn't be there I think because it's initialization code, it doesn't belong in a controller method

It's not the best solution. I wrote it before LoggerInterface was used in CronArchive; I think now we could supply a nother implementation.

> The fact that the custom init stuff is in this file was better I think because it reminds us that archive.php is just kept for BC and is a pain: it should be removed when we can. Custom stuff that's in there is hacky, and we should contain that to this file and not spread it elsewhere as much as we can.

I wanted to remove this script entirely and rely on the API method, however, according to @mattab this functionality will always be required and will never be removed.